### ISSUE TITLE
Need Help to Repeat YOLOv8n

### ISSUE THREAD

**Issue by wenhui-ml** [ultralytics/ultralytics#13146]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I try to repeat yolov8n train with the offcial training args using the following code.
"
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
print(model.ckpt['train_args'])
"
And the  offcial training args just as below
{'task': 'detect', 'mode': 'train', 'model': 'yolov8n.yaml', 'data': 'coco.yaml', 'epochs': 500, 'patience': 50, 'batch': 16, 'imgsz': 640, 'save': True, 'cache': 'disk', 'device': 0, 'workers': 8, 'project': 'YOLOv8', 'name': 'yolov8n', 'exist_ok': False, 'pretrained': False, 'optimizer': 'SGD', 'verbose': False, 'seed': 0, 'deterministic': True, 'single_cls': False, 'image_weights': False, 'rect': False, 'cos_lr': False, 'close_mosaic': 10, 'resume': False, 'overlap_mask': True, 'mask_ratio': 4, 'dropout': False, 'val': True, 'save_json': False, 'save_hybrid': False, 'conf': 0.001, 'iou': 0.7, 'max_det': 300, 'half': True, 'dnn': False, 'plots': True, 'source': 'ultralytics/assets/', 'show': False, 'save_txt': False, 'save_conf': False, 'save_crop': False, 'hide_labels': False, 'hide_conf': False, 'vid_stride': 1, 'line_thickness': 3, 'visualize': False, 'augment': False, 'agnostic_nms': False, 'retina_masks': False, 'format': 'torchscript', 'keras': False, 'optimize': False, 'int8': False, 'dynamic': False, 'simplify': False, 'opset': 17, 'workspace': 4, 'nms': False, 'lr0': 0.01, 'lrf': 0.01, 'momentum': 0.937, 'weight_decay': 0.001, 'warmup_epochs': 3.0, 'warmup_momentum': 0.8, 'warmup_bias_lr': 0.1, 'box': 7.5, 'cls': 0.5, 'dfl': 1.5, 'fl_gamma': 0.0, 'label_smoothing': 0.0, 'nbs': 64, 'hsv_h': 0.015, 'hsv_s': 0.7, 'hsv_v': 0.4, 'degrees': 0.0, 'translate': 0.1, 'scale': 0.5, 'shear': 0.0, 'perspective': 0.0, 'flipud': 0.0, 'fliplr': 0.5, 'mosaic': 1.0, 'mixup': 0.0, 'copy_paste': 0.0, 'cfg': None, 'v5loader': True}

And the below is my training args with ultralytics==8.2.10, which is the same as yolov8n  main training args
task: detect
mode: train
model: yolov8n.yaml
data: coco.yaml
epochs: 500
time: null
patience: 50
batch: 16
imgsz: 640
save: true
save_period: -1
cache: false
device: 0
workers: 8
project: null
name: yolov8n_offical_args
exist_ok: false
pretrained: false
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: false
val: true
split: val
save_json: false
save_hybrid: false
conf: 0.001
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.001
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml
save_dir: runs/detect/yolov8n_offical_args

**Comment by glenn-jocher** [comment]
Hi there,

Thanks for reaching out! It looks like you've done a thorough job in matching the official training arguments for YOLOv8n. To ensure everything is aligned, you might want to double-check a few key parameters such as `half`, `cache`, and `project` settings. 

Here’s a quick example to help you get started with the official training arguments:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')

# Train the model with the official arguments
results = model.train(
    data='coco.yaml',
    epochs=500,
    patience=50,
    batch=16,
    imgsz=640,
    save=True,
    cache='disk',
    device=0,
    workers=8,
    project='YOLOv8',
    name='yolov8n',
    optimizer='SGD',
    seed=0,
    deterministic=True,
    overlap_mask=True,
    mask_ratio=4,
    val=True,
    conf=0.001,
    iou=0.7,
    max_det=300,
    half=True,
    plots=True,
    lr0=0.01,
    lrf=0.01,
    momentum=0.937,
    weight_decay=0.001,
    warmup_epochs=3.0,
    warmup_momentum=0.8,
    warmup_bias_lr=0.1,
    box=7.5,
    cls=0.5,
    dfl=1.5,
    hsv_h=0.015,
    hsv_s=0.7,
    hsv_v=0.4,
    translate=0.1,
    scale=0.5,
    fliplr=0.5,
    mosaic=1.0
)
```

For more detailed guidance, you can refer to the [Ultralytics YOLO training documentation](https://docs.ultralytics.com/modes/train/). If you have any further questions, feel free to ask. Happy training! 🚀

**Comment by wenhui-ml** [comment]
Hi，is yolov8 scratch training with parameter of "multi_scale=True"?
According to the dicussuion above, I cannot get the same mAP50:95 37.3，but a little lower 36.5，just as below

**Comment by wenhui-ml** [comment]
yolo detect val model=yolov8n_ciou_offical_args.pt imgsz=640 data=coco.yaml conf=0.001 iou=0.7
all       5000      36781      0.616      0.477      0.514      0.361
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.365
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.520
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.396
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.178
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.400
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.518
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.315
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.526
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.581
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.352
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.642
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.774

**Comment by glenn-jocher** [comment]
Hello,

Thanks for sharing your validation results. Achieving an mAP of 36.5 is quite close to the expected 37.3, and such minor variations can often be attributed to factors like initialization, slight differences in training conditions, or even hardware used for training and validation.

Regarding your question about `multi_scale`: If you're aiming to replicate the official results, ensure that all training parameters, including `multi_scale`, match those used in the official training setup. Small discrepancies in settings can lead to variations in performance.

Keep experimenting with different configurations, and you might find tweaks that improve your results even further! 🚀 If you have more questions or need further assistance, feel free to ask.
